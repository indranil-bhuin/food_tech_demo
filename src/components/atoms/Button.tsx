import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';

const Button = ({ onPress, text, style, backgroundColor }) => {
  return (
    <TouchableOpacity style={[styles.button, style, { backgroundColor }]} onPress={onPress}>
      <Text style={styles.buttonText}>{text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    width: 320,
    height: 70,
    //marginTop: 150,
    backgroundColor: '#FFFFFF',
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonText: {
    fontSize: 16,
    fontWeight: 'bold',
    color: '#FFF'
  },
});

export default Button;
