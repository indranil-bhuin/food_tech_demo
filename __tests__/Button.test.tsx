import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/components/atoms/Button';

describe('Button', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <Button
        onPress={() => console.log('Button pressed!')}
        text="Test Button"
        style={{ marginBottom: 10 }}
        backgroundColor="#FFFFFF"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
