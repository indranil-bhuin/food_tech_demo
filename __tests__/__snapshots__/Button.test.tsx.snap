// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 25,
      "height": 70,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": 150,
      "opacity": 1,
      "width": 320,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Test Button
  </Text>
</View>
`;
